### PR TITLE
Break status into execution status and finality status

### PR DESCRIPTION
**Pull request by ArielElp**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/107)

### CODE REVIEW

File: api/starknet_api_openrpc.json [hunk 1]
```diff
@@ -2242,20 +2246,23 @@
...
+                        "title": "Finality status",
                         "type": "string",
                         "enum": [
-                            "ACCEPTED_ON_L2",
-                            "REVERTED"
+                            "ACCEPTED_ON_L2"
                         ],
-                        "description": "The status of the transaction"
+                        "description": "The finality status of the transaction"
+                    },
```
Comment (Mirko-von-Leipzig) [inline api/starknet_api_openrpc.json:2262]: This should be `TXN_FINALITY_STATUS` and not its own enum?

**Review by joshklop — COMMENTED** [review]
I think the `required` fields for `COMMON_RECEIPT_PROPTERTIES` need to be updated to reflect the changes.

**Review by ArielElp — COMMENTED** [review]
correct, done

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/107#-:-NZebSsp8Jb_gfiE_V46:bekut4r)* status: all files reviewed, 1 unresolved discussion (waiting on @EvyatarO and @Mirko-von-Leipzig)

___
*[`api/starknet_api_openrpc.json` line 2262 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/107#gh-1251943814:-NZe_pyJCojdOQPzqIzy:bjhgcfs) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/e9a99424c2fbe9fcccdc5d439658c0c62b9b5896/api/starknet_api_openrpc.json#L2262)):*
<details><summary><i>Previously, Mirko-von-Leipzig (Mirko von Leipzig) wrote…</i></summary><blockquote>

This should be `TXN_FINALITY_STATUS` and not its own enum?
</blockquote></details>

This is PENDING_RECEIPT_... so there’s only one possible option for the finality. So far, this is how we have represented "constant" fields in different objects. WDYT?

**Review by ArielElp — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> Dismissed @Mirko-von-Leipzig from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/107)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @EvyatarO)

**Review by EvyatarO — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/107)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ArielElp)

**Comment by yoga-braavos** [comment]
@ArielElp @EvyatarO , as the case with the gateway API, I suggest to keep the `status` for backwards compatibility reasons.

**Comment by EvyatarO** [comment]
@yoga-braavos 
The feeder gateway will keep the old `status` field. And the RPC v0.3.0 also has it.

**Pull request merged**